components/script/dom/bindings/structuredclone.rs

```diff
@@ -67,16 +67,49 @@ unsafe fn read_length(r: *mut JSStructuredCloneReader)
   return length as usize;
 }
 
+struct StructuredCloneWriter {
+    w: *mut JSStructuredCloneWriter,
+}
+
+impl StructuturedCloneWriter {
+    unsafe fn write_slice(&self, v: &[u8]) {
+        assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, v.len()));
```

Comment:
We should be writing the length before writing the bytes, so the calling code does not need to write the length manually.